Library/Homebrew/language/haskell.rb

```diff
@@ -92,5 +92,50 @@ def install_cabal_package(*args, **options)
         end
       end
     end
+
+    module CabalV2
+      def cabal_sandbox(options = {})
+        pwd = Pathname.pwd
+        home = options[:home] || pwd
+
+        # pretend HOME is elsewhere, so that ~/.cabal is kept as untouched
+        # as possible (except for ~/.cabal/setup-exe-cache)
+        # https://github.com/haskell/cabal/issues/1234
+        saved_home = ENV["HOME"]
+        ENV["HOME"] = home
+
+        system "cabal", "v2-update"
```

Comment:
I'm wondering if it's worth just having this `system` called directly instead of the sandbox method abstraction.